tests/test_graph.py

```diff
@@ -501,8 +502,9 @@ def test_function_graph_has_cycles_false():
     assert fg.has_cycles(nodes, user_nodes) is False
 
 
-def test_function_graph_display():
+def test_function_graph_display(tmp_path: pathlib.Path):
```

Comment:
This test was initially making two assertions:
1. is the content of the file valid
2. does it **not** create a file when passed `output_file_path = None`

I split this into two tests: `test_function_graph_display()` and `test_function_graph_display_no_dot_output()`